analyzers/src/SonarAnalyzer.CSharp/Rules/DisposableNotDisposed.cs

```diff
@@ -158,10 +158,12 @@ private static bool IsNodeInsideUsingStatement(SyntaxNode node)
         {
             var ancestors = node.AncestorsAndSelf().ToArray();
             var usingStatements = ancestors.OfType<UsingStatementSyntax>();
-            var usingDeclarations = ancestors.OfType<LocalDeclarationStatementSyntax>();
+            var usingDeclarations = ancestors
```

Comment:
looking again at the overall logic, I don't fully understand how this works.

Now I am looking at the `usingStatements.Any(x => ancestors.Contains(x.Expression) || ancestors.Contains(x.Declaration))` logic.

It's not verifying if the `node` is inside a using statement, but rather if, inside all the ancestors of the node (namespace, class, method, various blocks inside the method), it contains the expression or declaration inside the using statement.

This seems quite strange.

Reply:
For 

```
 await using (fs2 = new FileStream(@"c:\foo.txt", FileMode.Open)) // UsingStatement 
```

x.Expression will be 

```
fs2 = new FileStream(@"c:\foo.txt", FileMode.Open) // SimpleAssignmentExpression 
```

And `ancestors.Contains(x.Expression)` will always be true because the node itself is the expression.

This is a very convoluted way of expressing

```
ancestors.Contains(node)) 
```

instead of `usingStatements.Any`

because

`node.Equals(usingStatements.First().Expression)` will always be true.

Reply:
Knowing that a node is inside the using statement - by itself - isn't enough to say the object is properly disposed of.
In this example, a `FileStream `object is created inside the body Statement part of the using statement, which isn't exposed properly.
https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABADAAhQRgHQEkB5AbgFgAoFAZiwCYMBhDAbwo3axpQBYMBZABQBKFmw7j0GAQDcwAJwywAlhAAOAewgBTWIQBGAKy0BjYBgC8GAHZaA7hgBiSgDZaAysDlawAWwEAiPGAAD2B/aEcXLT51WC1cQlUtKyERMXF2VnIMnIxZBSt1YAARFQ1tXUMTM0sbeydXDy9fANpcELCIhujY+MTkoTJs3IBfdIwx8hGgA===

If the node is inside the `Declaration `or `Expression `of the `UsingStatement `then I consider it properly disposed of, otherwise, if it's inside the `Statement `part then I don't.

Maybe if I change the method name it will help. How about `IsInsideUsingDeclarationOrExpression`? Or `IsDisposedInUsingStatement`?